xla/tests/build_defs.bzl

```diff
@@ -168,12 +168,13 @@ def xla_test(
 
     test_names = []
     if not backends:
-        backends = _DEFAULT_BACKENDS
+        backends = _NVIDIA_DEFAULT_BACKENDS
 
     # Expand "gpu" backend name to specific GPU backends and update tags.
     backends, disabled_backends, backend_tags, backend_args = \
-        prepare_gpu_backend_data(backends, disabled_backends, backend_tags, backend_args)
+        prepare_nvidia_gpu_backend_data(backends, disabled_backends, backend_tags, backend_args)
 
+    backends += AMD_GPU_DEFAULT_BACKENDS
```

Comment:
This is adding the AMD default backends in all cases - which means the user can't overwrite it anymore. I think you rather have to add them in line 171.

Reply:
I need to do this after the call to prepare_nvidia_gpu_backend_data. Otherwise I need to compare for amd backend and continue.  I could not find a cleaner solution for this.